catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -264,22 +273,40 @@ public String getResponseWrapper(String responseType, Object response) {
 
       AttributeDescriptor descriptor = lookup.get();
       Map<String, Object> attributeProperties = new HashMap<>();
-      attributeProperties.put("type", descriptor.getType().getAttributeFormat().name());
-      attributeProperties.put("multivalued", descriptor.isMultiValued());
-      attributeProperties.put("id", descriptor.getName());
+      attributeProperties.put(TYPE_KEY, descriptor.getType().getAttributeFormat().name());
+      attributeProperties.put(MULTIVALUED_KEY, descriptor.isMultiValued());
+      attributeProperties.put(ID_KEY, descriptor.getName());
+      attributeProperties.put(ISINJECTED_KEY, true);
       Set<String> types =
           attribute.metacardTypes().isEmpty() ? resultTypes.keySet() : attribute.metacardTypes();
-      for (String type : types) {
-        Map<String, Object> attributes =
-            (Map) resultTypes.getOrDefault(type, new HashMap<String, Object>());
-        attributes.put(attribute.attribute(), attributeProperties);
-        resultTypes.put(type, attributes);
-      }
+
+      types
+          .stream()
+          .filter(type -> isAttributeMissing(resultTypes, attribute, type))
+          .forEach(
+              type -> {
+                Map<String, Object> attributes =
```

Comment:
✏️ might help readability extracting this method out to a separate method

Reply:
done